### PR TITLE
feat: pregenerated nodes for the tests

### PR DESCRIPTION
**Pull request by nicolasburtey**
cf: https://github.com/lightningnetwork/lnd/commit/e488bbfc9d6c406b955e791672f4993d20a9568d

### CODE REVIEW

**Comment by openoms** [comment]
Just to clarify the usecase will this be useful when restoring a node from seed will not need to share the macaroons again?

Might be related to remote signing to make sure the existing macaroons can access the signing node as well?

**Comment by nicolasburtey** [comment]
> Just to clarify the usecase will this be useful when restoring a node from seed will not need to share the macaroons again?
> 
> Might be related to remote signing to make sure the existing macaroons can access the signing node as well?

[until this PR that has being merged yesterday](https://github.com/lightningnetwork/lnd/commit/e488bbfc9d6c406b955e791672f4993d20a9568d), there was no way to say to a node "use this macaroon". the macaroon was generated when the node was running. 

Because of that, we have this script that is [fetching the macaroon](https://github.com/GaloyMoney/galoy/blob/5ca2774501dd11ca7d7db8d261775b8ad1bd59dd/.envrc#L11) dynamically every time we're doing tests.

this script takes maybe 5 seconds to execute. and we need to constantly run it, both in development, and when doing any tests on devnet that required lnd. 


so instead of having lnd generating the macaroon, if we could inject it, then the .envrc will run in milliseconds instead of seconds, which would both improve the development experience (main benefits) but reduce testing time on every jobs by 5 - 10 seconds. 

note: in addition to the macaroon, the seed/pubkey should also be injected, so that we can also get rid of [this function](https://github.com/GaloyMoney/galoy/blob/5ca2774501dd11ca7d7db8d261775b8ad1bd59dd/.envrc#L33). the ability to inject the seed was already there but because there was no way to inject the macaroon we didn't refactored our script because there would still need to have a `fetch_` function anyway.

**Comment by nicolasburtey** [comment]
note: @openoms I've put you as a reviewer because I thought this is something you might want to work on. feel free to pass it on to someone else if you'd like.

**Comment by openoms** [comment]
Right, thanks for the explanation. This is interesting, I'll dive in!

**Comment by openoms** [comment]
As a first step for the regtest environment will need to hardcode a `wallet_password`, `cipher_seed_mnemonic` and a `macaroon_root_key` instead of using `noseedbackup=true`.

Will use the https://api.lightning.community/?javascript#initwallet to initiate the wallet.

**Comment by openoms** [comment]
In this PR:
* adding pregenerated files for the 4 nodes on regtest used for the unit tests including the wallet.db, macaroons.db, admin.macaroon and the base64 representation of the admin.macaroon and tls.cert
* works on the current lnd as there are is no macaroon_root_key used
* these are copied to a .gitignored directory every time the `docker compose up` is used (this is because lnd modifies the files while mounted in the image)
* running test if there is a perceivable speed up, but nonetheless the generation of wallets and reading info from the docker containers is avoided

**Comment by nicolasburtey** [comment]
tests seems to be running 40s faster, for a 5m test. this is a great improvment!

File: dev/lnd/save-lnd-data.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (nicolasburtey) [reply dev/lnd/save-lnd-data.sh:1]: merging this. I'll let @openoms we can move it in another PR if there is a better place for it. I think it's only here to update static file which should not really happen

[the remark this answers is not in the document]
Comment (openoms) [reply dev/lnd/save-lnd-data.sh:1]: can be moved without consequence as it is only there as a reference to generate the saved data.
Are there any other scripts we should move? Can do in a separate PR

**Pull request merged**